packages/@aws-cdk/aws-location-alpha/lib/route-calculator.ts

```diff
@@ -0,0 +1,166 @@
+import * as iam from 'aws-cdk-lib/aws-iam';
+import { ArnFormat, IResource, Lazy, Resource, Stack, Token } from 'aws-cdk-lib/core';
+import { Construct } from 'constructs';
+import { CfnRouteCalculator } from 'aws-cdk-lib/aws-location';
+import { DataSource } from './place-index';
+import { generateUniqueId } from './util';
+
+/**
+ * A Route Calculator
+ */
+export interface IRouteCalculator extends IResource {
+  /**
+   * The name of the route calculator
+   *
+   * @attribute
+   */
+  readonly routeCalculatorName: string;
+
+  /**
+   * The Amazon Resource Name (ARN) of the route calculator resource
+   *
+   * @attribute Arn,CalculatorArn
+   */
+  readonly routeCalculatorArn: string;
+}
+
+/**
+ * Properties for a route calculator
+ */
+export interface RouteCalculatorProps {
+  /**
+   * A name for the route calculator
+   *
+   * Must be between 1 and 100 characters and contain only alphanumeric characters,
+   * hyphens, periods and underscores.
+   *
+   * @default - A name is automatically generated
+   */
+  readonly routeCalculatorName?: string;
+
+  /**
+   * Data source for the route calculator
+   *
+   * @default DataSource.ESRI
+   */
+  readonly dataSource?: DataSource;
+
+  /**
+   * A description for the route calculator
+   *
+   * @default - no description
+   */
+  readonly description?: string;
+}
+
+abstract class RouteCalculatorBase extends Resource implements IRouteCalculator {
```

Comment:
Same comment to another PR, don't think Base construct class is needed.

Reply:
Thanks. I've removed base class